src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -519,6 +525,20 @@ internal async Task LoadPackageMetadaAsync(IPackageMetadataProvider metadataProv
             }
         }
 
+        private async Task<IEnumerable<(VersionInfo versionInfo, PackageDeprecationMetadata deprecationMetadata)>> GetVersionsWithDeprecationMetadataAsync(IPackageMetadataProvider metadataProvider)
+        {
+            var versions = await _searchResultPackage.GetVersionsAsync();
```

Comment:
How many registration calls happen here?

Reply:
It depends.

In general, none--usually either the versions and deprecation will already be loaded by something else or the `VersionInfo` object was created a `GetVersionsAsync` and `GetDeprecationMetadataAsync` that are just `Task.FromResult` and return immediately.

The worst case scenario is when a package has many different versions across multiple feeds and the data is not already loaded. In that case, this function will make a call to each feed's registration once; however, the code already does this with the existing `GetVersionsAsync` call. The `GetDeprecationMetadataAsync` call, if it does go to registration, reuses the same registration call that the `GetVersionsAsync` call does.